tools/CMakeLists.txt

```diff
@@ -1,174 +1,131 @@
+    ${HEIR_EXTENSION_LIBS}
+    ${HEIR_CONVERSION_LIBS}
+    # ${HEIR_TRANSLATION_LIBS} # No need for translations
+
+    # Anything using add_mlir_library is not automatically included above
+
+    # For most things, there is a convenience target defined:
+    HEIRAnalysis # Anything from lib/Anaylysis folder
+    HEIRUtils # Anything from lib/Utils folder
+    HEIRSource # Anything from lib/Source folder
+    HEIRTransforms # Anything from lib/Transforms folder
+    # HEIRTarget # Anything from lib/Target folder
+
+    # That just leaves the dialect-specific transforms:
+    # TODO: Create an add_mlir_transform or similar function
+    # to avoid the need to manually list all dialect-specific transforms
+    HEIRBooleanVectorizer
+    HEIRLWETransforms
+    HEIROpenfheTransforms
+    HEIRPolynomialTransforms
+    HEIRSecretTransforms
+    HEIRSetDefaultParameters
+    HEIRTensorExtTransforms
+    HEIRTosaToSecretArith
+
+    # # TODO: Optimize CMake MLIR/LLVM dependencies to decrease tool binary sizes
+    ${MLIR_DIALECT_LIBS} # allows registerAllDialects()
+    ${MLIR_EXTENSION_LIBS} # allows registerAllExtensions()
+    ${MLIR_CONVERSION_LIBS} # allows registerAllPasses()
+    # ${MLIR_TRANSFORMS_LIBS} # mostly relevant for heir-translate
 )
-add_dependencies(${TARGET_NAME} MLIRPolynomialPassesIncGen)
-add_dependencies(${TARGET_NAME} MLIRPolynomialNTTRewritePassesIncGen)
-add_dependencies(${TARGET_NAME} MLIRTosaToSecretArith)
-add_dependencies(${TARGET_NAME} MLIRLinalgToTensorExt)
-endfunction()
 
-message(STATUS "Configuring heir-translate")
-make_heirtx_exec(heir-translate heir-translate.cpp)
+add_heir_executable(heir-translate heir-translate.cpp)
+target_link_libraries(heir-translate PRIVATE
+    # Utility Function used to define the main entry function
+    MLIRTranslateLib
 
-message(STATUS "Configuring heir-lsp")
-make_heirlsp_exec(heir-lsp heir-lsp.cpp)
+    # TODO: use more granualar dependency tracking
+    ${HEIR_DIALECT_LIBS}
+    ${HEIR_EXTENSION_LIBS}
+    # ${HEIR_CONVERSION_LIBS}
+    ${HEIR_TRANSLATION_LIBS}
 
-message(STATUS "Configuring heir-opt")
-make_heiropt_exec(heir-opt heir-opt.cpp)
+    # Anything using add_mlir_library is not automatically included above
+
+    # For most things, there is a convenience target defined:
+    # HEIRAnalysis # Anything from lib/Anaylysis folder
+    HEIRUtils # Anything from lib/Utils folder
+    HEIRSource # Anything from lib/Source folder
+    # HEIRTransforms # Anything from lib/Transforms folder
+    HEIRTarget # Anything from lib/Target folder
+
+    # That just leaves the dialect-specific transforms:
+    # TODO: Create an add_mlir_transform or similar function
+    # to avoid the need to manually list all dialect-specific transforms
+    # HEIRBooleanVectorizer
+    # HEIRLWETransforms
+    # HEIROpenfheTransforms
+    # HEIRPolynomialTransforms
+    # HEIRSecretTransforms
+    # HEIRSetDefaultParameters
+    # HEIRTensorExtTransforms
+    # HEIRTosaToSecretArith
```

Comment:
I think these can just go? heir-translate (and lsp) shouldn't  need any transforms

Reply:
Ah, yes, I left the commented-out ones for each tool to make it explicit what ISN'T being included, but I'm also happy to remove the comments!